torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -153,8 +153,10 @@ void ProcessGroupAgent::join() {
   //    effort to fix this problem).
   sync();
   int dst = (pg_->getRank() + 1) % pg_->getSize();
+  auto future = std::make_shared<FutureMessage>();
   enqueueSend(
-      SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)));
+      SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)),
+      future);
   threadPool_.waitWorkComplete();
   listenerThread_.join();
```

Comment:
We should add a future.wait() here and check for errors?

Reply:
@pritamdamania87, can you elaborate on this a bit? Should we check if an exception occured when sending a shutdown message to the other process, and if so, should we log that and continue with shutting down this process?